examples/tcp.js

```diff
@@ -20,9 +27,24 @@ client.connect(53, '8.8.8.8', function () {
 })
 
 client.on('data', function (data) {
-  console.log('Received response')
-  console.log(packet.streamDecode(data))
-  client.destroy() // kill client after server's response
+  console.log('Received response: %d bytes', data.byteLength)
+  if (response == null) {
+    if (data.byteLength > 1) {
+      const plen = buf.readUInt16BE(0)
+      expected_length = plen
+      if (plen < 12) {
+        throw new Error('below DNS minimum packet length')
+      }
```

Comment:
This looks to be a static check on the request packet that does not need to happen during response parsing. I'd move it out of the `data` event, maybe directly after encoding the packet